fmriprep/workflows/anatomical.py

```diff
@@ -74,10 +76,130 @@ def t1w_preprocessing(name='t1w_preprocessing', settings=None):
+            for t1w in t1w_list:
+                img = nib.load(t1w)
+                if all((img.shape == t1w_ref.shape,
+                        img.header.get_zooms() == t1w_ref.header.get_zooms())):
+                    t1w_outs.append(t1w_outs)
+
+            autorecon1_flags = [default_flags]
+            reconall_flags = [default_flags]
+            if hires:
+                autorecon1_flags.append('-hires')
+                reconall_flags.append('-hires')
+            return (t1w_outs, ' '.join(autorecon1_flags),
+                    ' '.join(reconall_flags))
+
+        recon_config = pe.Node(
+            niu.Function(
+                function=detect_inputs,
+                input_names=['t1w_list', 'default_flags'],
+                output_names=['t1w', 'autorecon1_flags', 'reconall_flags']),
+            name='ReconConfig',
+            run_without_submitting=True)
+        recon_config.inputs.default_flags = '-noskullstrip'
+
+        def bidsinfo(in_file):
+            from fmriprep.interfaces.bids import BIDS_NAME
+            match = BIDS_NAME.search(in_file)
+            params = match.groupdict() if match is not None else {}
+            return tuple(map(params.get, ['subject_id', 'ses_id', 'task_id',
+                                          'acq_id', 'rec_id', 'run_id']))
+
+        bids_info = pe.Node(
+            niu.Function(function=bidsinfo, input_names=['in_file'],
+                         output_names=['subject_id', 'ses_id', 'task_id',
+                                       'acq_id', 'rec_id', 'run_id']),
+            name='BIDSInfo',
+            run_without_submitting=True)
+
+        autorecon1 = pe.Node(
+            freesurfer.ReconAll(
+                directive='autorecon1',
+                openmp=nthreads,
+                parallel=True),
+            name='Reconstruction')
+        autorecon1.interface._can_resume = False
+        autorecon1.interface.num_threads = nthreads
+        if 'FREESURFER_SUBJECTS' in os.environ:
+            autorecon1.inputs.subjects_dir = os.getenv('FREESURFER_SUBJECTS')
+
+        def inject_skullstripped(subjects_dir, subject_id, skullstripped):
+            import os
+            import nibabel as nib
+            from nilearn.image import resample_to_img, new_img_like
+            from nipype.utils.filemanip import copyfile
+            mridir = os.path.join(subjects_dir, subject_id, 'mri')
+            t1 = os.path.join(mridir, 'T1.mgz')
+            bm_auto = os.path.join(mridir, 'brainmask.auto.mgz')
+            bm = os.path.join(mridir, 'brainmask.mgz')
+
+            img = nib.load(t1)
+            mask = nib.load(skullstripped)
+            bmask = new_img_like(mask, mask.get_data() > 0)
+            resampled_mask = resample_to_img(bmask, img, 'nearest')
+            masked_image = new_img_like(img, img.get_data() * resampled_mask.get_data())
+            masked_image.to_filename(bm_auto)
+
+            copyfile(bm_auto, bm, copy=True, use_hardlink=True)
```

Comment:
Brain masks are so small and hard links are not supported on all systems. Will this throw an exception if hard links are not supported?